src/GameConfiguration/index.ts

```diff
@@ -79,12 +106,86 @@ export default class GameConfiguration implements GameConfiguration {
   }
 
   update(visibleGameState: VisibleGameInformation): this {
+    const { grid, tiles } = this.revealed
+    const { passable, unknownObstacles, adjacencies, distances, cities, crowns } = this.hidden
+
+    function tileAt(rowIndex: number, colIndex: number): Tile | undefined {
+      const row = grid[rowIndex]
+      return row && row[colIndex]
+    }
+
+    function maybeAddAdjacency(tile: Tile, otherTile: Tile | undefined): void {
+      if (otherTile && passable.has(otherTile)) {
+        adjacencies.get(tile)!.add(otherTile)
+        adjacencies.get(otherTile)!.add(tile)
+      }
+    }
+
+    function markTileAsPassable(tile: Tile): void {
+      passable.add(tile)
+      distances.get(tile)!.set(tile, 0)
+      const { rowIndex, colIndex } = tile
+      const north = tileAt(rowIndex - 1, colIndex)
+      const south = tileAt(rowIndex + 1, colIndex)
+      const west  = tileAt(rowIndex, colIndex - 1)
+      const east  = tileAt(rowIndex, colIndex + 1)
+      maybeAddAdjacency(tile, north)
+      maybeAddAdjacency(tile, south)
+      maybeAddAdjacency(tile, west)
+      maybeAddAdjacency(tile, east)
+
+      // update distances to this new tile
+      for (const otherTile of passable) {
+        for (const adjacency of adjacencies.get(tile)!) {
+          for (const [otherTile, adjacencyDistance] of distances.get(adjacency)!.entries()) {
+          const possiblySmallerDistance = adjacencyDistance + 1 
+          const currentDistance = distances.get(tile)!.get(otherTile)
+          const performUpdate = possiblySmallerDistance < currentDistance
+          if(performUpdate) {
+              distances.get(tile)!.set(otherTile, possiblySmallerDistance)
+              distances.get(otherTile)!.set(tile, possiblySmallerDistance)             
+            }
+          }
+        }
+      }
+
+      // update other distances which may now be shorter because of this new tile
+      for (const firstTile of passable) {
```

Comment:
Maybe this can/should be done as adjacencies are discovered, let's take care of the other comments and see where we stand.